traffic_ops/traffic_ops_golang/federation_resolvers/federation_resolvers.go

```diff
@@ -0,0 +1,138 @@
+package federation_resolvers
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import "fmt"
+import "net/http"
+
+import "github.com/apache/trafficcontrol/lib/go-tc"
+import "github.com/apache/trafficcontrol/lib/go-util"
+
+import "github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/api"
+import "github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/dbhelpers"
+
+const insertFederationResolverQuery = `
+INSERT INTO federation_resolver (ip_address, type)
+VALUES ($1, $2)
+RETURNING federation_resolver.id,
+          federation_resolver.ip_address,
+          (
+          	SELECT type.name
+          	FROM type
+          	WHERE type.id = federation_resolver.type
+          ) AS type
+`
+
+const readQuery = `
+SELECT federation_resolver.id,
+       federation_resolver.ip_address,
+       federation_resolver.last_updated,
+       type.name AS type
+FROM federation_resolver
+LEFT OUTER JOIN type ON type.id = federation_resolver.type
```

Comment:
reason for LEFT OUTER here compared to just LEFT JOIN?

Reply:
They're the same; `OUTER` is optional, just more explicit.